android_jni/avifandroidjni/src/main/java/com/avif/android/AvifDecoder.java

```diff
@@ -0,0 +1,58 @@
+package com.avif.android;
+
+import android.graphics.Bitmap;
+import java.nio.ByteBuffer;
+
+/** An AVIF Decoder. AVIF Specification: https://aomediacodec.github.io/av1-avif/. */
+@SuppressWarnings("CatchAndPrintStackTrace")
+public class AvifDecoder {
+  static {
+    try {
+      System.loadLibrary("avif_android");
+    } catch (UnsatisfiedLinkError exception) {
+      exception.printStackTrace();
+    }
+  }
+
+  // This is a utility class and cannot be instantiated.
```

Comment:
[This is just a comment, not a request for change.]

It took me a while to understand what this comment means, because I didn't know what a utility class means in Java. (It is a class with only static methods.) This seems like an unusual design. It also means the getInfo() and decode() method cannot share the same libavif AVIFDecoder object.

Reply:
Utility classes are those that contain only static members.

Yes, while there is some scope to share the decoder between getInfo() and decode(), there is no compelling reason to do so as of now. There may be callers who will skip getInfo() and directly call decode(). So, in order to keep the API simple, we don't share any variables across functions for now. This can change in the future if we decide to expose more functions or some such.

Reply:
If I understand it correctly, a user of this class cannot skip getInfo() and directly call decode() unless the width and height of the image are already known.

Reply:
Yes, for example if you are decoding thumbnails of a fixed size or some such then the application will be able to allocate a large enough bitmap and call decode() directly.

Reply:
Thank you for the example. I think that use case is uncommon. It will be more common to call `getInfo()`, followed by `decode()`, so in the common case the "parse" operation will be done twice. (This is the case in your Glide integration.)

We can deal with this issue later.

Reply:
yes, i agree that this can be improved.

we can land this PR as it provides a good starting point. minor incremental features such as this will be easy to add in the future. :)